pandas/core/arrays/period.py

```diff
@@ -636,6 +636,20 @@ def to_timestamp(self, freq=None, how: str = "start") -> DatetimeArray:
         >>> idx.to_timestamp()
         DatetimeIndex(['2023-01-01', '2023-02-01', '2023-03-01'],
         dtype='datetime64[ns]', freq='MS')
+
+        The frequency will not be inferred if the index contains less than
+        three elements and values of index are not strictly monotonic:
```

Comment:
the way this is phrased it sounds like (pseudocode):
```
if len(index) < 3 and not index.is_monotonic():
    dont_infer_frequency(...)
```
Instead, I think it's more like
```
if len(index) < 3 or not index.is_monotonic():
    dont_infer_frequency(...)
```
?

How about
```
        The frequency will not be inferred if the index contains less than
        three elements, or if the values of index are not strictly monotonic:
```

Reply:
thanks, you are right, my wording isn't correct. I changed it as you suggested.